hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenRejectHandler.java

```diff
@@ -118,9 +142,135 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
+            @NonNull final HederaConfig hederaConfig) {
+
+        // Initialize stores and get the rejecting account
+        final var tokenStore = context.writableStore(WritableTokenStore.class);
+        final var accountStore = context.writableStore(WritableAccountStore.class);
+        final var nftStore = context.writableStore(WritableNftStore.class);
+        final var relStore = context.writableStore(WritableTokenRelationStore.class);
+
+        final var rejectingAccount =
+                getIfUsable(rejectingAccountID, accountStore, context.expiryValidator(), INVALID_ACCOUNT_ID);
+
+        // Prepare collections for processing
+        final var tokenTransferLists = new ArrayList<TokenTransferList>();
+        final var updatedFungibleTokenAllowances = new ArrayList<>(rejectingAccount.tokenAllowances());
+
+        final var allowancesIsEnabled = hederaConfig.allowancesIsEnabled();
+        for (final var rejection : rejections) {
+            if (rejection.hasFungibleToken()) {
+                processFungibleTokenRejection(
+                        rejection,
+                        rejectingAccountID,
+                        tokenTransferLists,
+                        tokenStore,
+                        relStore,
+                        updatedFungibleTokenAllowances,
+                        allowancesIsEnabled);
+            } else if (rejection.hasNft()) {
+                processNftRejectionAndRemoveSpenderAllowance(
+                        requireNonNull(rejection.nft()),
+                        rejectingAccountID,
+                        tokenTransferLists,
+                        nftStore,
+                        tokenStore,
+                        allowancesIsEnabled);
+            }
+        }
+
+        // Update the account with the new fungible token allowances
+        if (allowancesIsEnabled) {
+            final var updatedAccount = rejectingAccount
+                    .copyBuilder()
+                    .tokenAllowances(updatedFungibleTokenAllowances)
+                    .build();
+            accountStore.put(updatedAccount);
+        }
+        return tokenTransferLists;
+    }
+
+    private void processFungibleTokenRejection(
+            @NonNull final TokenReference rejection,
+            @NonNull final AccountID rejectingAccountID,
+            @NonNull final List<TokenTransferList> tokenTransferLists,
+            @NonNull final WritableTokenStore tokenStore,
+            @NonNull final WritableTokenRelationStore relStore,
+            @NonNull final List<AccountFungibleTokenAllowance> tokenAllowancesForRejectingAccount,
+            final boolean allowancesEnabled) {
+        final var token = getIfUsable(requireNonNull(rejection.fungibleToken()), tokenStore, PERMIT_PAUSED);
+        final var tokenId = requireNonNull(token.tokenId());
+
+        final var tokenRelation = getIfUsable(rejectingAccountID, tokenId, relStore, TokenRelValidations.PERMIT_FROZEN);
```

Comment:
We should use `getIfUsableForAliasedId` if we are allowing aliases here

Reply:
I've adjusted it :)